### DISCUSSION TITLE
How do you query Pandas DataFrames from DuckDB?

### DISCUSSION THREAD

**Discussion by Chuxiaof** [duckdb/duckdb#4451] [Q&A]
Hi there, I am working on another project, where we need to query a Pandas DataFrames from Rust without copying everything over. Could you give me a high-level overview of how this is implemented in DuckDB? For example, how do you access/parse the DataFrame data structure in Python from Rust? Thanks!

**Comment by Alex-Monahan** [comment]
DuckDB is written in C++, so I don't know how helpful it would be as an example!

**Reply by Chuxiaof** [reply]
I'm sure it will be helpful since Rust and C++ occupies roughly the same niche as programming languages. At the very least, your approach can help me build a mental model. Please share some details if you know them, or drop some links to the relevant part of the source tree!

**Reply by Mause** [reply]
Most of the code lives here: https://github.com/duckdb/duckdb/blob/master/tools/pythonpkg/src/pandas_scan.cpp
